app/com/linkedin/drelephant/spark/heuristics/ConfigurationHeuristic.scala

```diff
@@ -128,18 +135,29 @@ object ConfigurationHeuristic {
     lazy val serializer: Option[String] = getProperty(SPARK_SERIALIZER_KEY)
 
     lazy val serializerSeverity: Severity = serializer match {
-      case None => Severity.NONE
+      case None => Severity.MODERATE
       case Some(`serializerIfNonNullRecommendation`) => Severity.NONE
-      case Some(_) => serializerIfNonNullSeverityIfRecommendationUnmet
+      case Some(_) => DEFAULT_SERIALIZER_IF_NON_NULL_SEVERITY_IF_RECOMMENDATION_UNMET
+    }
+
+    lazy val isDynamicAllocationEnabled: Option[Boolean] = getProperty(SPARK_DYNAMIC_ALLOCATION_ENABLED).map(_.toBoolean)
+    lazy val isShuffleServiceEnabled: Option[Boolean] = getProperty(SPARK_SHUFFLE_SERVICE_ENABLED).map(_.toBoolean)
+
+    lazy val shuffleAndDynamicAllocationSeverity = (isDynamicAllocationEnabled, isShuffleServiceEnabled) match {
```

Comment:
I think it could be useful to have a brief comment here outlining the rationale. The reader could of course trace it back to that commit, but a comment is more reader-friendly in my opinion.